HW9/MyArrayList.cs

```diff
@@ -0,0 +1,425 @@
+            {
+                throw new InvalidOperationException();
+            }
+            T result = RemoveByIndexElement(0);
+
+            return result;
+        }
+
+        public T RemoveBackElement()
+        {
+            if (Length == 0)
+            {
+                throw new InvalidOperationException();
+            }
+            T result = RemoveByIndexElement(Length - 1);
+
+            return result;
+        }
+
+        public T RemoveByIndexElement(int index)
+        {
+            if (Length == 0)
+            {
+                throw new InvalidOperationException();
+            }
+            IEnumerable<T> temp = RemoveByIndexNElements(index, 1);
+            T[] result = temp.ToArray();
+
+            return result[0];
+        }
+
+        public IEnumerable<T> RemoveFrontNElements(int value)
+        {
+            if (Length == 0)
+            {
+                throw new InvalidOperationException();
+            }
+            IEnumerable<T> result = RemoveByIndexNElements(0, value);
+
+            return result;
+        }
+
+        public IEnumerable<T> RemoveBackNElements(int count)
+        {
+            if (Length==0)
+            {
+                throw new InvalidOperationException();
+            }
+            if (count > Length || count <= 0)
+            {
+                throw new ArgumentException();
+            }
+            IEnumerable<T> result = RemoveByIndexNElements(Length - count, count);
+
+            return result;
+        }
+
+        public IEnumerable<T> RemoveByIndexNElements(int index, int count)
+        {
+            if (Length == 0)
+            {
+                throw new InvalidOperationException();
+            }
+            if (index > Length || index < 0)
+            {
+                throw new IndexOutOfRangeException();
+            }
+            if ((index + count) > Length || count <= 0)
+            {
+                throw new ArgumentException();
+            }
+            
+            T[] result = new T[count];
+            int resultIndex = 0;
+            var endPosition = index + count;
+
+            for (int i = index; i < endPosition; i++)
+            {
+                result[resultIndex] = _array[i];
+                resultIndex++;
+            }
+            for (int i = endPosition, j = index;
+                i < Length;
+                i++, j++)
+            {
+                _array[j] = _array[i];
+            }
+
+            _currentCount -= count;
+
+            return result;
+        }
+
+        public int FirstIndexByValue(T value)
+        {
+            int index = -1;
+            for (int i = 0; i < Length; i++)
+            {
+                if (_array[i].CompareTo(value)==0)
+                {
+                    index = i;
+                    break;
+                }
+            }
+
+            return index;
+        }
+
+        public void Reverse()
+        {
+            T[] arrayTemp = new T[Length];
```

Comment:
Wrong solution. You can use `for` with `Length / 2` limit and swap first and last, then second and item before last etc.